src/mpid/ch4/netmod/ucx/ucx_init.h

```diff
@@ -113,7 +113,7 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
 #endif
 
     /* we reserve one bit to differentiate AMs from native messages */
-    *tag_ub = MPIDI_UCX_TAG_USABLE_BITS;
+    *tag_bits = MPID_TAG_BITS;
```

Comment:
I'm not understanding how this is reserving a bit. Obviously it seems to be working from the Jenkins results, but I don't quite get it.

Reply:
This is just assigning the `tag_bits`, it is not reserving anything. I should have removed the comment. The restriction is on the `tag_ub` which, apparently, I forgot to fix. UCX netmod should overwrite `tag_ub` to a smaller value to reserve the `AM` bit. Probably it is not showing up in Jenkins because the test that uses the `MPI_TAG_UB` is not integrated yet (?)

Reply:
`tag_ub` is calculated as a function of the number of `tag_bits` available. The device should be reporting how many tag bits are supported. It is up to the upper layer to determine the `tab_ub`. Can you combine the large tag value test with this PR? I don't think we should merge this without it.